src/registry/partners/index.tsx

```diff
@@ -0,0 +1,115 @@
+import {clsx} from 'clsx';
+import {useFormikContext} from 'formik';
+import {FormattedMessage} from 'react-intl';
+
+import FormFieldContainer from '@/components/FormFieldContainer';
+import HelpText from '@/components/forms/HelpText';
+import Tooltip from '@/components/forms/Tooltip';
+import type {RegistryEntry} from '@/registry/types';
+
+import PartnersPreview from './PartnersPreview';
+import ValueDisplay from './ValueDisplay';
+import isEmpty from './empty';
+import './index.scss';
+import getInitialValues from './initialValues';
+import {Affixes, BSN, DateOfBirth, Initials, LastName} from './subFields';
+import type {ManuallyAddedPartnerDetails} from './types';
+import getValidationSchema from './validationSchema';
+
+export interface FormioPartnersFieldProps {
+  componentDefinition: PartnersComponentSchema;
+}
+
+export const FormioPartnersField: React.FC<FormioPartnersFieldProps> = ({
+  componentDefinition: {key, label, description, tooltip},
+}) => {
+  const {setFieldValue, getFieldProps} = useFormikContext();
+  const {value: partners} = getFieldProps<ManuallyAddedPartnerDetails[] | PartnerDetails[]>(key);
+
+  const manuallyAddedPartner = partners.find(
+    (partner): partner is ManuallyAddedPartnerDetails =>
+      // Because `__addedManually` is only specified on `ManuallyAddedPartnerDetails`,
+      // we need to validate it explicitly as a boolean value to make the linter happy
+      '__addedManually' in partner && !!partner?.__addedManually
+  );
+  const canAddPartner = partners.length === 0;
+  const canEditPartner = manuallyAddedPartner !== undefined;
+  const showPartnersList = partners.length > 0 && !canEditPartner;
+
+  // Conform the business rules, if there is a manually added partner,
+  // it must be the first and only partner.
+  const subFieldNamePrefix = `${key}.0`;
+
+  const addPartner = () => {
+    setFieldValue(key, [
+      {
+        bsn: '',
+        initials: '',
+        affixes: '',
+        lastName: '',
+        dateOfBirth: '',
+        __addedManually: true,
+      } satisfies ManuallyAddedPartnerDetails,
+    ]);
+  };
+
+  const removePartner = () => {
+    setFieldValue(key, []);
+  };
+
+  return (
+    <fieldset className="openforms-fieldset">
+      <legend
+        className={clsx('openforms-fieldset__legend', {
+          'openforms-fieldset__legend--tooltip': !!tooltip,
+        })}
+      >
+        {label}
+        {tooltip && <Tooltip>{tooltip}</Tooltip>}
+      </legend>
+
+      {showPartnersList && <PartnersPreview partners={partners} />}
+      {canEditPartner && (
+        <FormFieldContainer>
+          <BSN namePrefix={subFieldNamePrefix} />
+          <Initials namePrefix={subFieldNamePrefix} />
+          <Affixes namePrefix={subFieldNamePrefix} />
+          <LastName namePrefix={subFieldNamePrefix} />
+          <DateOfBirth namePrefix={subFieldNamePrefix} />
+
+          <div>
+            <PrimaryActionButton hint="danger" onClick={() => removePartner()}>
```

Comment:
given that `removePartner` is so short, I'd inline this instead of using a different function for it in the body of the component, which tends to improve readability.